test/strategy/liquity/LiquityStrategy.spec.ts

```diff
@@ -0,0 +1,415 @@
+            constants.AddressZero,
+            stabilityPool.address,
+            lqty.address,
+            underlying.address,
+          ],
+          {
+            kind: 'uups',
+          },
+        ),
+      ).to.be.revertedWith('StrategyAdminCannotBe0Address');
+    });
+
+    it('reverts if stabilityPool is address(0)', async () => {
+      await expect(
+        upgrades.deployProxy(
+          LiquityStrategyFactory,
+          [
+            vault.address,
+            admin.address,
+            constants.AddressZero,
+            lqty.address,
+            underlying.address,
+          ],
+          {
+            kind: 'uups',
+          },
+        ),
+      ).to.be.revertedWith('StrategyStabilityPoolCannotBeAddressZero');
+    });
+
+    it('reverts if lqty is address(0)', async () => {
+      await expect(
+        upgrades.deployProxy(
+          LiquityStrategyFactory,
+          [
+            vault.address,
+            admin.address,
+            stabilityPool.address,
+            constants.AddressZero,
+            underlying.address,
+          ],
+          {
+            kind: 'uups',
+          },
+        ),
+      ).to.be.revertedWith('StrategyYieldTokenCannotBe0Address');
+    });
+
+    it('reverts if underlying is address(0)', async () => {
+      await expect(
+        upgrades.deployProxy(
+          LiquityStrategyFactory,
+          [
+            vault.address,
+            admin.address,
+            stabilityPool.address,
+            lqty.address,
+            constants.AddressZero,
+          ],
+          {
+            kind: 'uups',
+          },
+        ),
+      ).to.be.revertedWith('StrategyUnderlyingCannotBe0Address');
+    });
+
+    it('reverts if vault does not have IVault interface', async () => {
+      await expect(
+        upgrades.deployProxy(
+          LiquityStrategyFactory,
+          [
+            manager.address,
+            admin.address,
+            stabilityPool.address,
+            lqty.address,
+            underlying.address,
+          ],
+          {
+            kind: 'uups',
+          },
+        ),
+      ).to.be.revertedWith('StrategyNotIVault');
+    });
+
+    it('checks initial values', async () => {
+      expect(await strategy.isSync()).to.be.true;
+      expect(await strategy.hasRole(DEFAULT_ADMIN_ROLE, admin.address)).to.be
+        .true;
+      expect(await strategy.hasRole(MANAGER_ROLE, vault.address)).to.be.true;
+      expect(await strategy.vault()).to.eq(vault.address);
+      expect(await strategy.stabilityPool()).to.eq(stabilityPool.address);
+
+      expect(await strategy.underlying()).to.eq(underlying.address);
+
+      expect(await strategy.hasAssets()).to.be.false;
+      expect(
+        await underlying.allowance(strategy.address, stabilityPool.address),
+      ).to.eq(constants.MaxUint256);
+    });
+  });
+
+  describe('#transferAdminRights', () => {
+    it('can only be called by the current admin', async () => {
```

Comment:
I guess the test description here is that it fails when called by someone that doesn't have an admin role

Reply:
[fixed](https://github.com/lindy-labs/sc_solidity-contracts/pull/171/commits/ddb8aca1a9be2443bc3251bea91644ad215ef89b)